src/Lykke.Service.EasyBuy.DomainServices/OrdersService.cs

```diff
@@ -0,0 +1,197 @@
+            _settingsService = settingsService;
+            _tradesRepository = tradesRepository;
+            _instrumentsAccessService = instrumentsAccessService;
+            _log = logFactory.CreateLog(this);
+        }
+        
+        public async Task<Order> CreateAsync(string walletId, string priceId, decimal quotingVolume)
+        {
+            
+            var price = await _pricesService.GetAsync(priceId);
+            
+            if(price == null)
+                throw new EntityNotFoundException();
+
+            var instrument = await _instrumentsAccessService.GetByAssetPairIdAsync(price.AssetPair);
+            
+            if(instrument.State != InstrumentState.Active)
+                throw new FailedOperationException($"Instrument {instrument.AssetPair} isn't active.");
+
+            var defaultSettings = await _settingsService.GetDefaultSettingsAsync();
+            
+            if(DateTime.UtcNow > price.ValidTo + (instrument.OverlapTime ?? defaultSettings.OverlapTime))
+                throw new FailedOperationException("Given price too old.");
+            
+            if(quotingVolume > price.QuotingVolume)
+                throw new FailedOperationException("Requested volume higher than initial.");
+
+            var pair = await _assetsService.TryGetAssetPairAsync(price.AssetPair);
+
+            var baseAsset = await _assetsService.TryGetAssetAsync(pair.BaseAssetId);
+
+            var baseVolume = quotingVolume == price.QuotingVolume
+                ? price.BaseVolume
+                : (quotingVolume / price.Value).TruncateDecimalPlaces(baseAsset.Accuracy, price.Type == OrderType.Sell);
+            
+            var order = new Order
+            {
+                Id = Guid.NewGuid().ToString(),
+                WalletId = walletId,
+                Type = price.Type,
+                AssetPair = price.AssetPair,
+                QuotingVolume = quotingVolume,
+                BaseVolume = baseVolume,
+                PriceId = priceId,
+                CreatedTime = DateTime.UtcNow,
+                Status = OrderStatus.New,
+                ReserveTransferId = Guid.NewGuid().ToString(),
+                SettlementTransferId = Guid.NewGuid().ToString()
+            };
+
+            await _ordersRepository.InsertAsync(order);
+            
+            _log.Info("Order was created.", order);
+
+            try
+            {
+                await _internalTransfersService.TransferAsync(
+                    order.ReserveTransferId,
+                    walletId,
+                    await _settingsService.GetWalletIdAsync(),
+                    order.Type == OrderType.Buy
+                        ? pair.QuotingAssetId
+                        : pair.BaseAssetId,
+                    order.Type == OrderType.Buy
+                        ? quotingVolume
+                        : baseVolume);
+
+            }
+            catch (MeNotEnoughFundsException)
```

Comment:
It's better to avoid usage of special exception - this could be handled using MeOperationException.StatusCode in catch (MeOperationException e) block below

Reply:
In the domain layer, we don't care about infrastructure. The specific exception, in this case, will be better. The status code could be used in a specific service that communicated with ME.